docs/post-deployment-actions.md

```diff
@@ -0,0 +1,269 @@
+# Post-Deployment Actions
+
+It's common for a project to need to execute transactions after a deployment has occurred. This guide will show you how to do that with Sphinx.
+
+## Table of Contents
+
+- [Standard usage](#standard-usage)
+- [Calling external contracts](#calling-external-contracts)
+- [Chain-specific addresses](#chain-specific-addresses)
+- [Chain-specific function arguments](#chain-specific-function-arguments)
+- [Using contract references](#using-contract-references)
+- [Executing permissioned actions](#executing-permissioned-actions)
+  - [OpenZeppelin's `Ownable.sol`](#openzeppelins-ownablesol)
+  - [OpenZeppelin's `AccessControl.sol`](#openzeppelins-accesscontrolsol)
+
+## Standard usage
+
+Say you're deploying a contract called `MyContract`, and you'd like to call a function on it called `increment()` after it's deployed. You can do this by updating your config file to include the following:
+
+```ts
+// Import Sphinx types.
+import { Contract, UserSphinxConfig } from '@sphinx-labs/core'
+
+// Create a new Contract object for 'MyContract'. The contract reference `{{ MyContract }}` resolves
+// to the address of 'MyContract'.
+const MyContract = new Contract('{{ MyContract }}')
+
+const config: UserSphinxConfig = {
+  // Your existing config fields:
+  projectName: ...
+  contracts: { MyContract: { ... } },
+  // Define your post-deployment actions:
+  postDeploy: [
+    // Call the `increment()` function on `MyContract`:
+    MyContract.increment()
+  ]
+}
+```
+
+The `increment()` function will be called on `MyContract` for every chain that you deploy to.
+
+It's important to note that the post-deployment actions are executed after **all** of the contracts in the config file have been deployed.
+
+You can put as many function calls as you'd like in the `postDeploy` array. For example:
+```ts
+postDeploy: [
+  MyContract.increment(),
+  MyContract.setMyValue(2),
+  MyContract.transferOwnership('0x1234...')
+]
+```
+
+The post-deployment actions are executed in the order they're defined in the `postDeploy` array. So, in the example above, the order would be: `increment`, `setMyValue`, then `transferOwnership`.
+
+Like contract deployments, your post-deployment actions are idempotent. This means that they'll only be executed once per chain, even if you re-deploy your config file.
```

Comment:
Nit: It would be nice to provide a bit more detail here. I.e explaining that if the input arguments change for a specific chain, then we will treat that as a new action and it will get executed.